crates/ika-core/src/dwallet_mpc/crytographic_computation/mpc_computations/presign.rs

```diff
@@ -4,66 +4,259 @@
+                    .secp256r1_protocol_public_parameters()?;
+                PresignPublicInputByProtocol::Secp256r1ECDSA(generate_presign_public_input::<
+                    Secp256R1ECDSAProtocol,
+                >(
+                    session_identifier,
+                    protocol_public_parameters,
+                    dwallet_public_output,
+                )?)
+            }
         };
 
-        Ok(pub_input)
+        Ok(input)
+    }
+}
+
+fn generate_presign_public_input<P: Protocol>(
+    session_identifier: SessionIdentifier,
+    protocol_public_parameters: P::ProtocolPublicParameters,
+    dwallet_public_output: Option<SerializedWrappedMPCPublicOutput>,
+) -> DwalletMPCResult<<PresignParty<P> as mpc::Party>::PublicInput> {
+    let dkg_output = dwallet_public_output
+        .clone()
+        .ok_or(DwalletMPCError::MPCSessionError {
+            session_identifier,
+            error: "presign public input cannot be None as we only support ECDSA".to_string(),
+        })?;
+    let dkg_output: VersionedDwalletDKGSecondRoundPublicOutput =
+        bcs::from_bytes(dkg_output.as_slice())?;
+
+    let decentralized_dkg_output = match dkg_output {
+        VersionedDwalletDKGSecondRoundPublicOutput::V1(output) => {
+            bcs::from_bytes::<P::DecentralizedPartyDKGOutput>(output.as_slice())?
+        }
+        VersionedDwalletDKGSecondRoundPublicOutput::V2(output) => {
+            bcs::from_bytes::<P::DecentralizedPartyDKGOutput>(output.as_slice())?
+        }
+    };
+    let pub_input: <PresignParty<P> as mpc::Party>::PublicInput =
+        (protocol_public_parameters, decentralized_dkg_output).into();
+
+    Ok(pub_input)
+}
+
+pub fn compute_presign<P: Protocol>(
+    party_id: PartyID,
+    access_structure: &WeightedThresholdAccessStructure,
+    session_id: CommitmentSizedNumber,
+    advance_request: AdvanceRequest<<P::PresignParty as mpc::Party>::Message>,
+    public_input: <P::PresignParty as mpc::Party>::PublicInput,
+    rng: &mut impl CsRng,
+) -> DwalletMPCResult<GuaranteedOutputDeliveryRoundResult> {
+    let result =
+        mpc::guaranteed_output_delivery::Party::<P::PresignParty>::advance_with_guaranteed_output(
+            session_id,
+            party_id,
+            access_structure,
+            advance_request,
+            None,
+            &public_input,
+            rng,
+        )
+        .map_err(|e| DwalletMPCError::FailedToAdvanceMPC(e.into()))?;
+
+    match result {
+        GuaranteedOutputDeliveryRoundResult::Advance { message } => {
+            Ok(GuaranteedOutputDeliveryRoundResult::Advance { message })
+        }
+        GuaranteedOutputDeliveryRoundResult::Finalize {
+            public_output_value,
+            malicious_parties,
+            private_output,
+        } => {
+            // Wrap the public output with its version.
+            let public_output_value =
+                bcs::to_bytes(&VersionedPresignOutput::V2(public_output_value))?;
```

Comment:
Technically during this PR you didn't need `V2` at all, the presign remained the same for ECDSA, for Schnorr it just generated a new non-existing struct so still `V1`.

In the next universal/global presign PR, we do need to make it `v2` for ECDSA as the structure changed.
We can keep it `v1` for Schnorr - we are going to deserialize the structure differently anyways based on whether its schnorr or ECDSA, and there is only one way to deserialize schnorr (hence v1) and two for ecdsa (hence v2)

in this pr, please revert this change, let's do it properly as part of the universal presign pr